### PR TITLE
Add alpha to SetNameplateColor

### PR DESCRIPTION
**Pull request by attickdoor**
I inserted a few alpha values of 1 where hardcoded values were set. Other than that, I *think* this should just work. Existing calls that omit the alpha argument shouldn't break.

### CODE REVIEW

File: Plater.lua [hunk 1]
```diff
@@ -4355,7 +4355,7 @@ function Plater.OnInit() --private --~oninit ~init
 			if (plateFrame.PlateConfig.healthbar_color_by_hp) then
 				local originalColor = plateFrame.PlateConfig.healthbar_color
 				local r, g, b = DF:LerpLinearColor (abs (currentHealth / currentHealthMax - 1), 1, originalColor[1], originalColor[2], originalColor[3], 1, .4, 0)
-				Plater.ChangeHealthBarColor_Internal (self, r, g, b, true)
+				Plater.ChangeHealthBarColor_Internal (self, r, g, b, originalColor[4], true)
```
Comment (cont1nuity) [inline Plater.lua:4358]: may need nil-check here or in ChangeHealthBarColor_Internal (see comments there)
`(originalColor[4] or 1)` should do

File: Plater.lua [hunk 2]
```diff
@@ -4551,19 +4551,19 @@ end
 	end
 	
 	--internal function to change the health bar color
-	function Plater.ChangeHealthBarColor_Internal (healthBar, r, g, b, forceNoLerp) --private
-		if (r ~= healthBar.R or g ~= healthBar.G or b ~= healthBar.B) then
-			healthBar.R, healthBar.G, healthBar.B = r, g, b
+	function Plater.ChangeHealthBarColor_Internal (healthBar, r, g, b, a, forceNoLerp) --private
```
Comment (cont1nuity) [inline Plater.lua:4554]: need to keep an eye on this: adding a nil-check for alpha here might be useful as well to counter potential issues with profiles without the alpha value in the colors. this should not happen, as the default values are adjusted and the options store the alpha as well, but one never knows...

**Comment by cont1nuity** [comment]
Looks like a thorough change. Need to do a deeper review of the changes and see if something misses, but it has potential to be pulled for alpha after the next release.

**Comment by Yiwozai** [comment]
Good feature, I just need this.

**Comment by attickdoor** [comment]
Updated to address both concerns. 

> Looks like a thorough change. Need to do a deeper review of the changes and see if something misses, but it has potential to be pulled for alpha after the next release.

I may do similar for other elements as well (namely cast bar). But this is definitely the biggest one to get in.

File: Plater.lua [hunk 3]
```diff
@@ -4552,6 +4552,9 @@ end
 	
 	--internal function to change the health bar color
 	function Plater.ChangeHealthBarColor_Internal (healthBar, r, g, b, a, forceNoLerp) --private
+		if (a == nil) then
```
Comment (cont1nuity) [inline Plater.lua:4555]: `a = a or 1` please

File: Plater.lua [hunk 4]
```diff
@@ -4355,7 +4355,7 @@ function Plater.OnInit() --private --~oninit ~init
 			if (plateFrame.PlateConfig.healthbar_color_by_hp) then
 				local originalColor = plateFrame.PlateConfig.healthbar_color
 				local r, g, b = DF:LerpLinearColor (abs (currentHealth / currentHealthMax - 1), 1, originalColor[1], originalColor[2], originalColor[3], 1, .4, 0)
-				Plater.ChangeHealthBarColor_Internal (self, r, g, b, originalColor[4], true)
+				Plater.ChangeHealthBarColor_Internal (self, r, g, b, originalColor[4] or 1, true)
```
Comment (cont1nuity) [inline Plater.lua:4358]: `(originalColor[4] or 1)` with the parenthesis for better readability please

**Comment by cont1nuity** [comment]
> 
> 
> Will it be better that the alpha keep original instead of 1 when the paramter is nil?

You might be right about that. leaving alpha to `nil` might be best

**Comment by attickdoor** [comment]
> You might be right about that. leaving alpha to `nil` might be best in some situations. But currently default is 1 anyways and when explicitly setting the color having it default to 1 is fine as well.

Would `a = a or healthBar.A or 1` work?

**Comment by cont1nuity** [comment]
> 
> 
> > You might be right about that. leaving alpha to `nil` might be best in some situations. But currently default is 1 anyways and when explicitly setting the color having it default to 1 is fine as well.
> 
> Would `a = a or healthBar.A or 1` work?

No, as ` healthBar.A` can be removed, actually. It does not provide anything. The other values stored are for color-transitions and alpha is not used in this.

**Comment by cont1nuity** [comment]
Congrats on the PR :)

**Pull request merged**